src/smartAuthorizationHelper.ts

```diff
@@ -45,6 +45,18 @@ function isRequestorReferenced(
     fhirVersion: FhirVersion,
 ): boolean {
     const sourceResourceType = sourceResource.resourceType;
+    // this particularly for only detectedIssue resource because our matrix doesn't include cases for extension
+    if (sourceResourceType === 'DetectedIssue' && 'extension' in sourceResource) {
```

Comment:
Hardcoding is done here as resourceReferencesMatrixV4 defined in this repo doesn't include extention as possiblePaths to look the reference. Additionally, the extension object needs some filtering to lookout for the organization reference. Thus url is hardcoded to filter that extension object value.

Reply:
Why don't want we update the resrouceRefererencesMatrixV4 JSON instead. Update the "DetectedIssue" object in the JSON to include the following field:
`"Organization":["extension.valueReference"]`
It's a bit of hack because it doesn't discriminate which extension elements to check for the value, but if it avoids introducing custom code like this then I think it is a better approach.

Reply:
For example [ImagingStudy](https://www.hl7.org/fhir/imagingstudy.html) has an entry `series.performer.actor`, where both `series` and `performer` are array types, and `performer` is optional.

Reply:
awesome. You were right Lee. I have removed the changes in `isRequestorReferenced` function and added `"Organization":["extension.valueReference"]` in resrouceRefererencesMatrixV4 JSON.